TranslatorReasonerAPI.yaml

```diff
@@ -784,10 +797,67 @@ components:
           items:
             $ref: '#/components/schemas/Attribute'
           nullable: true
+        qualifiers:
+          type: array
+          description: >-
+            A set of Qualifiers that act together to add nuance
+            or detail to the statement expressed in an Edge.
+          items:
+            $ref: '#/components/schemas/Qualifier'
+          nullable: true
       additionalProperties: false
       required:
         - subject
         - object
+    Qualifier:
+      additionalProperties: false
+      description: >-
+        An additional nuance attached to an assertion
+      type: object
+      properties:
+        qualifier_type_id:
+          type: string
+          description: >-
+            The category of the qualifier, drawn from a hierarchy of qualifier
+            slots in the Biolink model (e.g. subject_aspect, subject_direction,
+            object_aspect, object_direction, etc).
+          example: subject_aspect
+          nullable: false
+        qualifier_value:
+          type: string
+          description: >-
+            The value associated with the type of the qualifier, drawn from
+            a set of controlled values by the type as specified in
+            the Biolink model (e.g. 'expression' or 'abundance' for the
+            qualifier type 'subject_aspect', etc).
+          example: expression
+          nullable: false
+      required:
+        - qualifier_type_id
+        - qualifier_value
+    QualifierConstraint:
+      additionalProperties: false
+      description: >-
+        Defines a query constraint based on the
+        qualifier_types and qualifier_values of a set of
+        Qualifiers attached to an edge. For example, it can constrain a
+        "ChemicalX - affects - ?Gene" query to return only edges where
+        ChemicalX specifically affects the 'expression' of the Gene,
+        by constraining on the qualifier_type "object_aspect" with
+        a qualifier_value of "expression".
+      properties:
+        qualifier_set:
+          type: array
+          description: >-
+            A set of Qualifiers that serves to add nuance to a query,
+            by constraining allowed values held by Qualifiers
+            on queried Edges.
+          items:
+            $ref: '#/components/schemas/Qualifier'
+          nullable: false
+      required:
+        - qualifiers
```

Comment:
@edeutsch , shouldn't `required` field in `QualifierConstraint` contain `qualifier_set` instead of `qualifiers` ?

Reply:
yes, absolutely, good catch! Fix in PR with https://github.com/NCATSTranslator/ReasonerAPI/pull/330/commits/f4bcfc532d80e990b30bfc8294c8fecd2e7e5f1d
thanks!